### PR TITLE
Modularise message display in templates

### LINKED ISSUE #152: Popup notifications don't show up on all pages
The popup notifications for messages like "submitted successfully" or "your draft has been saved" don't show up on all dashboard pages. This means that the pile up, and display all at once when the user loads a page that does support them.

(Note that this refers to the popup notifications for temporary messages, not the more permanent notifications that show up in the user menu and under "notifications" in your profile).

The templates need to be refactored so that such notifications show up on all pages—either by editing `base.html` or by adding the code to all relevant dashboard pages. The latter option could be implemented more easily if we make one template snippet and include it in relevant places (eg. `{% include 'dashboard/templates/snippets/show_commets.html' %}`).

### PR DESCRIPTION
**Pull request by badrihippo**
Made reusable code to render messages and added it on all relevant pages. Fixes #152.

### CODE REVIEW

File: dashboard/templates/dashboard/articles/review.html [hunk 1]
```diff
@@ -8,6 +8,8 @@
 {% block content %}
 <div class="review-page-container">
     <div class="answer-area">
+        {% include 'snippets/show_messages.html' %}
+        {% endif %}
```
Comment (nitinpaul) [inline dashboard/templates/dashboard/articles/review.html]: This seems to be a stray `endif`. Can you check?

File: sawaliram_auth/templates/sawaliram_auth/base.html [hunk 2]
```diff
@@ -16,11 +16,12 @@
     </head>
     <body>
         <div class="container auth-container">
+            {% include 'snippets/show_messages.html' %}
```
Comment (nitinpaul) [inline sawaliram_auth/templates/sawaliram_auth/base.html]: Messages are not shown really in auth views. I think we can stick to the form error messages with these for now!

File: dashboard/templates/dashboard/articles/review.html (hunk 1 above)
Comment (badrihippo) [reply dashboard/templates/dashboard/articles/review.html]: Fixed

File: sawaliram_auth/templates/sawaliram_auth/base.html (hunk 2 above)
Comment (badrihippo) [reply sawaliram_auth/templates/sawaliram_auth/base.html:19]: They might be at some point. No harm in leaving it there.

**Comment by jr-sawaliram** [comment]
@badrihippo is this code still usable? Else how do we address #152?

**Comment by badrihippo** [comment]
The main `snippets/show_messages.html` template is still usable. However the usage in the 13 conflicting files will need to be reimplemented, because those files have been modified in other ways since this PR was made.

It should be pretty straightforward to resolve though. People basically need to remove any legacy message-displaying code (identifiable by `{% if messages %}` or `{% for message in messages %}`) and instead use the line

     {% include 'snippets/show_messages.html' %}

to display messages on that page. This line should ideally show up on *all* pages, in an appropriately chosen place. Once this happens, all it takes is a line in the relevant code to show popup messages for various actions.

**Comment by jr-sawaliram** [comment]
Fixed by #381.

**Pull request closed without merging**